custom_components/tryfi/pytryfi/__init__.py

```diff
@@ -1,106 +1,187 @@
+                sentry_sdk.init(
+                        SENTRY_URL,
+                        release=PYTRYFI_VERSION,
+                    )
+            self._api_host = API_HOST_URL_BASE
+            self._session = requests.Session()
+            self._user_agent = "pyTryFi"
+            self._username = username
+            self._password = password
+            self.login()
+            #set Headers only after login for use going forward.
+            self.setHeaders()
+
+            self._currentUser = FiUser(self._userId)
+            self._currentUser.setUserDetails(self._session)
+
+            petListJSON = query.getPetList(self._session)
+            h = 0
+            for house in petListJSON:
+                for pet in petListJSON[h]['household']['pets']:
+                    #If pet doesn't have a collar then ignore it. What good is a pet without a collar!
+                    if pet['device'] != "None":
+                        p = FiPet(pet['id'])
+                        p.setPetDetailsJSON(pet)
+                        #get the current location and set it
+                        pLocJSON = query.getCurrentPetLocation(self._session,p._petId)
+                        p.setCurrentLocation(pLocJSON)
+                        #get the daily, weekly and monthly stats and set
+                        pStatsJSON = query.getCurrentPetStats(self._session,p._petId)
+                        p.setStats(pStatsJSON['dailyStat'],pStatsJSON['weeklyStat'],pStatsJSON['monthlyStat'])
+                        #get the daily, weekly and monthly rest stats and set
+                        pRestStatsJSON = query.getCurrentPetRestStats(self._session,p._petId)
+                        p.setRestStats(pRestStatsJSON['dailyStat'],pRestStatsJSON['weeklyStat'],pRestStatsJSON['monthlyStat'])
+                        #get behavior stats if available (Series 3+ only)
+                        try:
+                            p.updateBehaviorStats(self._session)
+                        except Exception as e:
+                            LOGGER.debug(f"Behavior stats not available for {p._name} - likely not a Series 3+ collar: {e}")
+                        LOGGER.debug(f"Adding Pet: {p._name} with Device: {p._device._deviceId}")
+                        self._pets.append(p)
+                    else:
+                        LOGGER.warning(f"Pet {pet['name']} - {pet['id']} has no collar. Ignoring Pet!")
+                h = h + 1
+
+            baseListJSON = query.getBaseList(self._session)
+            h = 0
+            for house in baseListJSON:
+                for base in baseListJSON[h]['household']['bases']:
+                    b = FiBase(base['baseId'])
+                    b.setBaseDetailsJSON(base)
+                    LOGGER.debug(f"Adding Base: {b._name} Online: {b._online}")
+                    self._bases.append(b)
+                h = h + 1
+        except Exception as e:
+            capture_exception(e)
```

Comment:
In a library, this style of error handling is not great because it means callers don't actually know when errors happen. We should let this bubble up and only handle errors that can reasonably be handled in a library.